### PR TITLE
Move ballista standalone mode to client

### LINKED ISSUE #588: [Ballista] Create "self-contained" mode in client library
From https://github.com/ballista-compute/ballista/issues/637

It would be nice to have a truly self-contained mode for Ballista, where you could run some queries without needing any additional processes. This would be an opt-in feature of the client crate.

Right now, you need to launch an executor in "standalone mode" (i.e. with an in-proc scheduler), which is not ideal. Once we add this, we could remove the "standalone mode" in the executor.

See [this](https://github.com/ballista-compute/ballista/pull/630) for context

### PR DESCRIPTION
**Pull request by edrevo**
# Which issue does this PR close?

Closes #588.

# What changes are included in this PR?
- Standalone feature in client lib
- Removed --local param in executor

# Are there any user-facing changes?
Yes, the executor no longer has a --local mode.

### CODE REVIEW

**Comment by edrevo** [comment]
cc @andygrove

**Review by andygrove — APPROVED** [review]

File: dev/docker/ballista-base.dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 
 # Base image extends debian:buster-slim
-FROM rust:1.51.0-buster AS builder
+FROM rust:1.52.1-buster AS builder
```
Comment (Jimexist) [inline dev/docker/ballista-base.dockerfile:26]: 1.53 is out now

**Review by houqp — APPROVED** [review]

**Comment by edrevo** [comment]
Build failure seems like a Cargo issue. I've opened https://github.com/rust-lang/cargo/issues/9608. Let's see if they confirm it is a bug or not.

File: dev/docker/ballista-base.dockerfile (hunk 1 above)
Comment (edrevo) [reply dev/docker/ballista-base.dockerfile:26]: i have 2 more PRs open. i'll change it in one of the other ones, if that's ok

**Comment by edrevo** [comment]
@andygrove, this one should be ready for merge

**Pull request merged**